wannierberri/__fermisea2.py

```diff
@@ -54,55 +54,75 @@ def SpinTot(data,Efermi):
 factor_ohmic=(elementary_charge/Ang_SI/hbar**2  # first, transform to SI, not forgeting hbar in velocities - now in  1/(kg*m^3)
                  *elementary_charge**2*TAU_UNIT  # multiply by a dimensional factor - now in A^2*s^2/(kg*m^3*tau_unit) = S/(m*tau_unit)
                    * 1e-2  ) # now in  S/(cm*tau_unit)
+
 def conductivity_ohmic(data,Efermi):
     return IterateEf(data.Ohmic,data,Efermi,TRodd=False,Iodd=False)*factor_ohmic
 
 def gyrotropic_Kspin(data,Efermi):
     factor_Kspin=-bohr_magneton/Ang_SI**2   ## that's it!
     return IterateEf(data.gyroKspin,data,Efermi,TRodd=False,Iodd=True)*factor_Kspin
 
-
 def Morb(data,Efermi, evalJ0=True,evalJ1=True,evalJ2=True):
     fac_morb =  -eV_au/bohr**2
-#    return Omega_tot(data,Efermi).mul_array(Efermi)
-    return  fac_morb*( 
-               IterateEf(data.Hplus(),data,Efermi,TRodd=True,Iodd=False) 
-                           -2*Omega_tot(data,Efermi).mul_array(Efermi) )*data.cell_volume
-
+    return  fac_morb*(
+                    IterateEf(data.Hplus(),data,Efermi,TRodd=True,Iodd=False)
+                            -2*Omega_tot(data,Efermi).mul_array(Efermi) )*data.cell_volume
 
-def tensor_D(data,Efermi):
-    return IterateEf(data.derOmegaTr,data,Efermi,TRodd=False,Iodd=True)
-
-def Hplus(data,Efermi):
-    return IterateEf(data.derHplusTr,data,Efermi,TRodd=False,Iodd=True)
+def HplusTr(data,Efermi):
+    return IterateEf(data.derHplusTr,data,Efermi,sep=True,TRodd=False,Iodd=True)
 
 def tensor_K(data,Efermi):
-    Hp = Hplus(data,Efermi).data
+    Hp = HplusTr(data,Efermi).data
     D = tensor_D(data,Efermi).data
     tensor_K = - elementary_charge**2/(2*hbar)*(Hp - 2*Efermi[:,None,None]*D  )
     return result.EnergyResult(Efermi,tensor_K,TRodd=False,Iodd=True)
 
+def tensor_D(data,Efermi):
+        return IterateEf(data.derOmegaTr,data,Efermi,sep=True,TRodd=False,Iodd=True)
 #########################
 ####  Private part ######
 #########################
 
 
-def IterateEf(dataIO,data,Efermi,TRodd,Iodd,rank=None,kwargs={}):
+def IterateEf(dataIO,data,Efermi,TRodd,Iodd,sep=False,rank=None,kwargs={}):
     """ this is a general function which accepts dataIO  -- a dictionary like {'i':i , 'io':io, ...}
 and sums for a series of Fermi levels"""
 #    funname=inspect.stack()[1][3]
 #    print ("iterating function '{}' for Efermi={}".format(funname,Efermi))
-    OCC=OccDelta(data.E_K,dataIO)
-    RES=[OCC.evaluate(Ef) for Ef in  Efermi ]
-    return result.EnergyResult(Efermi,np.cumsum(RES,axis=0)/(data.NKFFT_tot*data.cell_volume),TRodd=TRodd,Iodd=Iodd,rank=rank)
+# parameter deltaIO can be a dictionary or a funciton. If needed use callable(dataIO) for judgment and run OCC=OccDelta(data.E_K,dataIO(op,ed),op,ed) or OCC=OccDelta(data.E_K(op,ed),dataIO(op,ed),op,ed)
```

Comment:
Instead of comment write a docstring. Thus it may be automatically included in the documentation. also `deltaIO `-> `dataIO`